Microsoft.AVS.VVOLS/Microsoft.AVS.VVOLS.psm1

```diff
@@ -0,0 +1,370 @@
+    }
+
+    Write-Host "Certificates refreshed successfully for all of ESXi hosts."
+}
+
+<#
+    .SYNOPSIS
+     Create a new VASA provider in vCenter
+
+    .PARAMETER ProviderName
+     Name of VASA provider to be created in vCenter
+
+    .PARAMETER FlashArrayMgmtIP
+     URL of the VASA provider service
+
+    .PARAMETER ProviderCredential
+     Credential of the VASA provider service
+
+    .EXAMPLE
+     New-VvolVasaProvider -ProviderName "myProvider" -FlashArrayMgmtIP "1.0.0.0" -ProviderCredential $ProviderCredential
+#>
+function New-VvolVasaProvider {
+    [CmdletBinding()]
+    [AVSAttribute(10, UpdatesSDDC = $false)]
+    Param (
+        [Parameter(
+            Mandatory = $true,
+            HelpMessage = 'Name of VASA provider to be created in vCenter')]
+        [ValidateNotNull()]
+        [String]
+        $ProviderName,
+
+        [Parameter(
+            Mandatory = $true,
+            HelpMessage = 'URL of the VASA provider service')]
+        [ValidateNotNull()]
+        [String]
+        $ProviderUrl,
+
+        [Parameter(
+            Mandatory = $true,
+            HelpMessage = 'Credential of the VASA provider service')]
+        [ValidateNotNull()]
+        [PSCredential]
+        $ProviderCredential
+    )
+
+    $VasaProvider = Get-VasaProvider -Name $ProviderName -ErrorAction Ignore
+    if ($vasaProvider) {
+        throw "Unable to create a VASA provider. Vasa provider '$ProviderName' already exists."
+    }
+
+    New-VasaProvider -Name $ProviderName -Credential $ProviderCredential -Url $ProviderUrl -Force -ErrorAction Stop | Out-Null
+
+    Write-Host "VASA provider $ProviderName created successfully."
+}
+
+<#
+    .SYNOPSIS
+     Remove a VASA provider from vCenter
+
+    .PARAMETER ProviderName
+     Name of VASA provider to be removed from vCenter
+
+    .EXAMPLE
+     Remove-VvolVasaProvider -ProviderName "myProvider"
+
+    .INPUTS
+     vCenter VASA provider name
+
+    .OUTPUTS
+     None.
+#>
+function Remove-VvolVasaProvider {
+    [CmdletBinding()]
+    [AVSAttribute(10, UpdatesSDDC = $false)]
+    Param (
+        [Parameter(
+            Mandatory = $true,
+            HelpMessage = 'Name of VASA provider to be created in vCenter')]
+        [ValidateNotNull()]
+        [String]
+        $ProviderName
+    )
+
+    $VasaProvider = Get-VasaProvider -Name $ProviderName -ErrorAction Ignore
+    if (-not $VasaProvider) {
+        throw "Unable to remove a VASA provider. Vasa provider '$ProviderName' does not exist."
+    }
+
+    Remove-VasaProvider -Provider $VasaProvider -Confirm:$false -ErrorAction Stop | Out-Null
+
+    Write-Host "VASA provider $ProviderName removed successfully."
+}
+
+
+<#
+    .SYNOPSIS
+     Creates a new vVol Storage Policy
+
+    .PARAMETER PolicyConfigJsonString
+     Storage policy in Json string. The command will traverse through the SPBM rules and creat a storage policy containing all of the rules.
+     The example string is as below:
```

Comment:
Let's not specify the example in the param description. Please move it to `.EXAMPLE`.

Reply:
Done